### ISSUE TITLE
Consider retiring the Keras→TensorFlow conversion in `interop/` in favor of `importNetworkFromKeras`

### ISSUE THREAD

**Issue by stevevanhooser** [MATLAB-Community-Toolboxes-at-INCF/DeepInterpolation-MATLAB#92]
Splitting this out from the PR #91 discussion so it doesn't stay buried there.

Today the flagship pretrained model is a Keras `.h5` file that we round-trip through Python + TF 2.10 (`interop/save_model_tf.py`) into a TensorFlow SavedModel, because MATLAB's TF importer historically capped at TF 2.10. With [`importNetworkFromKeras`](https://www.mathworks.com/help/deeplearning/ref/importnetworkfromkeras.html) supporting TensorFlow 2.16+ (and available in R2025a, which is the minimum release we're targeting per PR #91), that intermediate step may no longer be needed — for future models published by the originator in Keras, and possibly for the current `.h5` as well.

Open questions:

- Can the existing `.h5` be loaded directly by `importNetworkFromKeras`, or does its Keras version pre-date what the new importer accepts? (The source model may not carry forward cleanly.)
- If it works for future zoo models but not the current one, is that enough to remove the Python detour going forward and leave the archival `.h5` on the old path?

If it pans out: retire `interop/save_model_tf.py` and the associated conversion step, update `+deepinterp` loading paths, and note the change in the README.

Cross-ref: PR #91.

**Comment by stevevanhooser** [comment]
Some upfront research from a read of the code, framed as a proposal for whoever picks this up.

Current code path (main today)

Three distinct import paths live in [pretrainedModels/pretrained.json](https://github.com/MATLAB-Community-Toolboxes-at-INCF/DeepInterpolation-MATLAB/blob/main/pretrainedModels/pretrained.json), only two of which are Keras-adjacent:

format: "TensorFlowZip" (four ophys/ephys models, TP-Ai93-450 included). Each is a zipped TensorFlow SavedModel produced by [interop/save_model_tf.py](https://github.com/MATLAB-Community-Toolboxes-at-INCF/DeepInterpolation-MATLAB/blob/main/interop/save_model_tf.py) under Python 3.9 + TF 2.10. deepinterp.Net('TensorFlowZip', …) → +deepinterp/+internal/openTensorFlowNetwork.m → importNetworkFromTensorFlow.
format: "KerasFile" (Allen_Training_And_Inference, a .h5). deepinterp.Net('KerasFile', …) → +deepinterp/+internal/importKerasMAE.m, which uses the legacy importKerasLayers, then findPlaceholderLayers / replaceLayer with a local maeRegressionLayer, then assembleNetwork.
format: "Matlab" (fMRI .mat). Unaffected.

So this issue actually touches two branches: the Python detour for the TensorFlowZip models, and the older importKerasLayers path for the KerasFile model.

Relevant importNetworkFromKeras facts (verify against the live doc page)
Introduced R2024a; R2025a target is comfortably in range.
Accepts Keras .h5 and .keras files.
Requires the Deep Learning Toolbox Converter for TensorFlow Models support package — same one the repo already asks users to install.
Returns a dlnetwork directly (not a LayerGraph, no assembleNetwork step).
Unsupported layers still surface as PlaceholderLayers, replaceable with findPlaceholderLayers + replaceLayer.
A dlnetwork has no terminal loss/output layer, so maeRegressionLayer is not needed for inference on this path — inference doesn't touch loss.
Older importKerasLayers / importKerasNetwork are documented as removed/not-recommended in current releases.
Downstream code that would care
deepinterp.Net.SetInputSize reads obj.network.Layers(1).InputSize — dlnetwork exposes .Layers, so this keeps working. Worth confirming the first layer is still the input layer with the same shape.
deepinterp.Net.interp calls predict(obj.network, …) — predict supports dlnetwork. If the network expects a specific dlarray format (e.g. "SSCB"), the current numeric input may need wrapping. Small parity test would settle it.
maeRegressionLayer.m becomes dead code on the new Keras path once verified.
Proposed changes (doc-only, no MATLAB run behind them)

A. New unified Keras path. Rewrite +deepinterp/+internal/importKerasMAE.m as a thin wrapper:

function net = importKerasMAE(kerasFile)
    net = importNetworkFromKeras(kerasFile);   % returns dlnetwork
    ph = findPlaceholderLayers(net);
    if ~isempty(ph)
        for k = 1:numel(ph)
            net = replaceLayer(net, ph(k).Name, ...
                deepinterp.internal.maeRegressionLayer);
        end
    end
end

To verify empirically: (a) does importNetworkFromKeras accept the specific .h5 in pretrainedModels/ (Keras-version compatibility), and (b) do any PlaceholderLayers actually appear — since a dlnetwork has no output layer, the historical MAE placeholder likely disappears, which would let the findPlaceholderLayers block and maeRegressionLayer.m be removed.

B. Retire the Python detour for the four TensorFlowZip models. Two sub-options:

B1 (minimally invasive): Keep the .zip files as-is; openTensorFlowNetwork.m continues to use importNetworkFromTensorFlow. Nothing in this issue forces removing them if they already work.
B2 (full retirement): Change those pretrained.json entries from "format": "TensorFlowZip" (pointing at .zip) to "format": "KerasFile" (pointing at the upstream .h5), update the URLs, and let the new unified Keras path handle them. Then [interop/save_model_tf.py](https://github.com/MATLAB-Community-Toolboxes-at-INCF/DeepInterpolation-MATLAB/blob/main/interop/save_model_tf.py), openTensorFlowNetwork.m, pretrainedModels/TensorFlowNetworks/, and the 'TensorFlowZip' case in Net.m's switch can all be removed. This is the version that fully answers the issue.

C. Net.m switch cleanup. If B2, the case 'tensorflowzip' branch goes away and the dispatch shrinks to 'kerasfile', 'matlab', and the still-stubbed 'new'.

D. Docs. README's "Deep Learning Toolbox Converter for TensorFlow Models" mention stays (the new importer still needs it), but the "Breakdown of DeepInterpolation TensorFlow model import" section becomes a Keras breakdown, and the corresponding example (tiny_ophys_inference_detailed) loses the Python step and shrinks by a couple of cells.

What documentation alone can't answer
Whether the specific .h5 files load cleanly via importNetworkFromKeras in R2025a (Keras-version compatibility).
Whether the ophys .h5 sources upstream at Allen Institute are still fetchable (current pretrained.json only has zip URLs).
Whether numeric parity holds between the old TensorFlow-imported network and the new Keras-imported one on the sample data — a quick interp() on [sampleData/ophys_tiny_761605196.tif](https://github.com/MATLAB-Community-Toolboxes-at-INCF/DeepInterpolation-MATLAB/blob/main/sampleData/ophys_tiny_761605196.tif) and a max-abs-diff would settle it.

All three want MATLAB R2025a + Deep Learning Toolbox + the TensorFlow converter locally to answer.
